Document asking Promptless to update your configuration in Slack/Teams (#782)
Promptless/docs: src/content/docs/docs/get-the-most-out/ask-promptless-to-update-config.mdx

Reviewer: Promptless doesn't support DMs on any channel.

Reviewer: If you want to gate edit access to the agent knowledge base via conversational interfaces, add an allow list in PROMPTLESS.md so only specified usernames/handles are allowed to make edits. Promptless will enforce this at runtime.

---
title: Ask Promptless to update your configuration
description: Ask Promptless in Slack or Microsoft Teams to change your triggers, doc collections, and other promptless.yaml settings directly
slug: docs/get-the-most-out/ask-promptless-to-update-config
type: guide
tags:
  - get-the-most-out
  - guide
timestamp: "2026-07-22T00:00:00Z"
sidebar:
  hidden: false
  order: 35
  label: Ask to update config
---

import { Aside } from '@astrojs/starlight/components';

> You can ask Promptless in Slack or Microsoft Teams to change your configuration, and it makes the change directly. This is a conversational alternative to editing `promptless.yaml` yourself on the Configuration page—describe the change you want in chat, and Promptless applies it.

Reviewer: ```suggestion
You can ask Promptless in Slack or Microsoft Teams to change your configuration, and it makes the change directly. This is a conversational alternative to editing `promptless.yaml` yourself on the Configuration page. Describe the change you want in chat, and Promptless applies it.
```

- **Triggers** — add, modify, or remove a trigger, such as a [`github_pr` trigger](/docs/reference/configuration-reference#triggers) scoped to a repo, a `slack_listen` trigger for a channel, or a commit trigger.
- **Doc collections** — add, remove, or rescope a [doc collection](/docs/reference/configuration-reference#doc-collections).
- **Context sources** — add or remove a [context source](/docs/reference/configuration-reference#context-sources).
- **Policies** — adjust publishing and notification rules, such as the [notification channel or `auto_create_pr`](/docs/reference/configuration-reference#policies).

These are common examples—you can change any setting `promptless.yaml` supports, not just these. For the full field reference, see the [Configuration reference](/docs/reference/configuration-reference).

## How to ask

@mention Promptless (or DM or message it) in Slack or Microsoft Teams the same way you already trigger doc updates, then phrase a natural-language request describing the config change you want.

> <Aside type="note">Microsoft Teams support is in beta.</Aside>

Reviewer: ```suggestion
```

For how to interact with the bot in Slack and Teams (message actions, mentions, and DMs), see [Review from Slack and Teams](/docs/work-the-queue/reviewing-from-slack-and-teams), and the [Slack](/docs/reference/integrations/slack) and [Microsoft Teams](/docs/reference/integrations/microsoft-teams) integration pages.

## Example requests

> These are illustrative—you can describe the change in your own words rather than following a fixed command syntax. The same requests work in Microsoft Teams; the `@Promptless` phrasing here isn't Slack-only.

Reviewer: ```suggestion
These are illustrative—you can describe the change in your own words rather than following a fixed command syntax. The same requests work in Slack and Microsoft Teams. The `@Promptless` phrasing here isn't tool-specific.
```

**Triggers**

- `@Promptless add a github_pr trigger for the acme/api repo`
- `@Promptless stop watching the #random channel`
- `@Promptless watch all repos in the acme org for docs changes`
- `@Promptless only trigger on the docs/ directory in acme/api`

**Doc collections**

- `@Promptless add the acme/docs repo as a doc collection`
- `@Promptless limit the acme/docs collection to the docs/ folder`

**Context sources**

- `@Promptless remove the Notion context source`

**Policies**

- `@Promptless send suggestion notifications to #docs-review`

> - `@Promptless turn on auto-create PRs for the acme/docs collection`

Reviewer: Use complete sentences and punctuation. We're a documentation product and need to appeal to our audience's sensibilities.

---
title: Ask Promptless to update your configuration
description: Ask Promptless in Slack or Microsoft Teams to change your triggers, doc collections, and other promptless.yaml settings directly
slug: docs/get-the-most-out/ask-promptless-to-update-config
type: guide
tags:
  - get-the-most-out
  - guide
timestamp: "2026-07-22T00:00:00Z"
sidebar:
  hidden: false
  order: 35
  label: Ask to update config
---

import { Aside } from '@astrojs/starlight/components';

> You can ask Promptless in Slack or Microsoft Teams to change your configuration, and it makes the change directly. This is a conversational alternative to editing `promptless.yaml` yourself on the Configuration page. Describe the change you want in chat, and Promptless applies it.

Reviewer: ```suggestion
You can ask Promptless in Slack or Microsoft Teams to change your configuration, and it makes the change directly. This is a conversational alternative to editing your configuration yourself in the Promptless dashboard. Describe the change you want in chat, and Promptless applies it.
```

## What you can change

All Promptless configuration lives in a single `promptless.yaml` file in your [Agent Knowledge Base](/docs/get-the-most-out/agent-knowledge-base), organized into four sections. Asking in chat can edit any of them:

- **Triggers** — add, modify, or remove a trigger, such as a [`github_pr` trigger](/docs/reference/configuration-reference#triggers) scoped to a repo, a `slack_listen` trigger for a channel, or a commit trigger.
- **Doc collections** — add, remove, or rescope a [doc collection](/docs/reference/configuration-reference#doc-collections).
- **Context sources** — add or remove a [context source](/docs/reference/configuration-reference#context-sources).
- **Policies** — adjust publishing and notification rules, such as the [notification channel or `auto_create_pr`](/docs/reference/configuration-reference#policies).

These are common examples—you can change any setting that `promptless.yaml` supports, not just these. For the full field reference, see the [Configuration reference](/docs/reference/configuration-reference).

## How to ask

> @mention Promptless in Slack or Microsoft Teams the same way you already trigger doc updates, then phrase a natural-language request describing the configuration change you want.

Reviewer: ```suggestion
Mention `@Promptless` in Slack or Microsoft Teams the same way you already trigger doc updates, then phrase a natural-language request describing the configuration change you want.
```

## Who can make changes

By default, anyone who can @mention Promptless in a connected Slack or Microsoft Teams channel can ask it to make configuration changes. This is different from editing configuration on the [Configuration page](/docs/reference/configuration-reference#configuration-editor), which is limited to admins.

To restrict who can make changes, add an allowlist of usernames or handles in `PROMPTLESS.md` in your [Agent Knowledge Base](/docs/get-the-most-out/agent-knowledge-base), alongside `promptless.yaml`. When an allowlist is present, Promptless enforces it at runtime, so only the listed users can make configuration changes through chat.

## Edit configuration yourself

You can also edit `promptless.yaml` directly on the Configuration page in [Form or YAML mode](/docs/reference/configuration-reference#configuration-editor).

<Aside type="note">Editing configuration on the dashboard is admin-only.</Aside>

> 

Reviewer: ```suggestion
```

To pair a trigger with written workflow instructions for a repeatable custom task, see [Teach Promptless a custom task](/docs/get-the-most-out/teach-promptless-a-custom-task).

> Need help? Contact us at [help@gopromptless.ai](mailto\:help@gopromptless.ai).

Reviewer: ```suggestion
```
